src/write/elf/object.rs

```diff
@@ -640,6 +641,7 @@ impl<'a> Object<'a> {
             (Architecture::Sbf, None) => elf::EM_SBF,
             (Architecture::Sharc, None) => elf::EM_SHARC,
             (Architecture::Sparc, None) => elf::EM_SPARC,
+            (Architecture::Sparc32Plus, None) => elf::EM_SPARC,
```

Comment:
```suggestion
            (Architecture::Sparc32Plus, None) => elf::EM_SPARC32PLUS,
```

You'll need this for the tests to pass. I'm surprised this didn't cause a problem for your use too, unless the linker allows combining objects with both machine types?

Reply:
Apologies, this was an oversight and a result of a copy-and-paste error.